### DISCUSSION TITLE
Invalid pin 20 for I2C on Adafruit ESP32 Feather V2

### DISCUSSION THREAD

**Discussion by mitchb1** [micropython/micropython#9372] [Hardware & Peripherals] [answered]
I am trying to make an I2C interface for Adafruit ESP32 Feather V2 (ESP32-PICO-MINI-02). On this board SCL is pin 20 and SDA is pin 22. When I try to make a pin object with Pin(20), I get an "invalid pin" ValueError. It looks like the cause of this is that pin 20 is listed as NULL in [micropython/ports/esp32.](https://github.com/micropython/micropython/blob/673db939b5c1f42c9411f684343d39811b7a16b2/ports/esp32/machine_pin.c#L162).

Maybe I am not specifying the board correctly somewhere or ?

**Update**: Maybe I'm confusing the chip pin number and the board pin number. On the [board schematic](https://cdn-learn.adafruit.com/assets/assets/000/109/518/original/adafruit_products_FV2_sch.png?1646779441), it shows that the chip pins are 34 and 26 for SDA and SCL. When I try those (i.e., I2C(1, scl=machine.Pin(26), sda=machine.Pin(34), freq=400000)), I get different errors:

E (350) i2c: /home/micropython/esp-idf-v4.2/components/driver/i2c.c:762 (i2c_set_pin):sda gpio number error
E (1360) i2c: /home/micropython/esp-idf-v4.2/components/driver/i2c.c:774 (i2c_set_pin):scl and sda gpio numbers are the same

**Comment by mitchb1** [comment, marked as answer]
Looks like this is a known [issue](https://learn.adafruit.com/adafruit-esp32-feather-v2/micropython-setup).

"Pin 20 does not currently work properly in MicroPython. This warning will remain until the necessary updates are completed by MicroPython to resolve this issue."

**Reply by jimmo** [reply]
See #9375 for more info. (Please don't raise duplicate posts)
